### PR TITLE
Make pixel perfect member directory screen from Figma in React

### PR DESCRIPTION
**Pull request by dbizhev**
## Purpose:
- Make pixel perfect member directory screen from Figma in React
image

## Freedcamp
- [Task 40026588](https://freedcamp.com/view/2956183/tasks/40026588)

### CODE REVIEW

**Review by ngvtuan — COMMENTED** [review]
UI issues 1

UI issues 2
`2` What happens if the user has a long name? Currently, when the name is too long it will be drop to the new line but in the existing design I think the text will be cut off instead of `drop to the new line`.

File: src/features/members/MemberStrip.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import React from 'react'
+
+export type MemberCard = {
+  avatar: string
+  name: string
+  followers: number | string
+  role?: string
+}
+
+export interface MemberStripProps {
```
[the remark this answers is not in the document]
Comment (dbizhev) [reply src/features/members/MemberStrip.tsx:10]: @Dinisueldanov ok but why though?

[the remark this answers is not in the document]
Comment (dbizhev) [reply src/features/members/MemberStrip.tsx:10]: @Dinisueldanov that was Nam, not Max

**Review by bulatsalyakhovdev — APPROVED** [review]

File: src/features/members/MemberStrip.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  memberCard: MemberCard
+  heighestSold: number | string
+  totalSell: number | string
+  images: string[]
+  // onFollowClick: () => void
```
[the remark this answers is not in the document]
Comment (dbizhev) [reply src/features/members/MemberStrip.tsx]: Fixed

File: src/features/members/MemberStrip.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      <div className='w-4/12 md:w-244px'>
+        <MemberCard {...memberCard} />
+      </div>
+      <div className='w-8/12 md:flex-grow border-b border-navigation-background'>
+        <div className='flex space-x-4 h-full'>
+          {/* Sale data */}
+          <div className='flex flex-col justify-center h-full'>
+            <h6 className='text-gray-77 pb-1 text-12'>Highest sold</h6>
+            <div className='w-max text-12 font-semibold border py-1 px-2 rounded-4px border-success-pressed text-success-pressed'>
+              {heighestSold} PSL
```
[the remark this answers is not in the document]
Comment (dbizhev) [reply src/features/members/MemberStrip.tsx]: Fixed

File: src/features/members/MemberStrip.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (dbizhev) [reply src/features/members/MemberStrip.tsx]: Fixed

File: src/features/members/MemberStrip.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
+import React from 'react'
+
+export type MemberCard = {
```
[the remark this answers is not in the document]
Comment (dbizhev) [reply src/features/members/MemberStrip.tsx]: Fixed

**Comment by dbizhev** [comment]
@ngvtuan we agreed to do the following at the moment:
1. Names can be cut.
2. Put the prices to a separate line.

>2 What happens if the user has a long name? Currently, when the name is too long it will be drop to the new line but in the existing design I think the text will be cut off instead of drop to the new line.

**Comment by dbizhev** [comment]
@ngvtuan fixed all UI issues you pointed out.

File: src/features/members/MembersDirectory.tsx [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+                  onChange={setRange}
+                  formatValue={formatValue}
+                  formatTooltipValue={formatValue}
+                />
+              </div>
+            </div>
+          </div>
+          <div className='space-y-5'>
+            {mockMemberStrips.map(item => (
+              <MemberStrip {...item} />
```
Comment (ngvtuan) [inline src/features/members/MembersDirectory.tsx:143]: Missing `key` prop

File: src/legacy/Routes.tsx [hunk 6]
```diff
@@ -649,7 +652,7 @@ class RouteApp extends React.Component<any, any> {
                     this.setRPCConfig(rpcConfig)
 
                     // set pastel DB thread update timer
-                    if (!app?.isPackaged) {
+                    if (process.env.NODE_ENV === 'production') {
```
Comment (ngvtuan) [inline src/legacy/Routes.tsx:655]: I think this code is not correct, We need to use `app.isPackaged`. See sample here: https://github.com/pastelnetwork/pastel-electron-wallet/blob/master/src/features/loading/LoadingScreen.tsx (Line 49)

File: src/legacy/Routes.tsx [hunk 7]
```diff
@@ -111,7 +112,7 @@ class RouteApp extends React.Component<any, any> {
     this.rpc = rpc
 
     // Auto refresh every 10s
-    if (!app?.isPackaged) {
+    if (process.env.NODE_ENV === 'production') {
```
Comment (ngvtuan) [inline src/legacy/Routes.tsx:115]: I think this code is not correct, We need to use app.isPackaged. See sample here: https://github.com/pastelnetwork/pastel-electron-wallet/blob/master/src/features/loading/LoadingScreen.tsx (Line 49)

**Review by ngvtuan — COMMENTED** [review]
error 1
Can we fix the above error? 

ui-issues 1
Base on this design I think we just need to add the scrollbar inside the content

ui-issues 2
`1` Not the same as the design 
`2` Color is not the same as the design. 
`3` Can we make the horizontal scrolling bar like the vertical scrolling bar?

**Comment by ArthurVogelsang** [comment]
@dbizhev font-family does not match with design
image

**Comment by ArthurVogelsang** [comment]
@dbizhev  doesn't match height & space
image

**Comment by dbizhev** [comment]
@ArthurVogelsang fixed UI issues you pointed out.

**Comment by dbizhev** [comment]
@ngvtuan fixed all the issues you pointed yesterday.

**Comment by ArthurVogelsang** [comment]
@dbizhev I think 1px is different between left line & horizontal scrollbar
image

**Review by ArthurVogelsang — CHANGES_REQUESTED** [review]

File: src/features/members/MemberStrip.tsx [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+                <span className='text-gradient'>{totalSell}</span>
+                <span className='text-gradient'> {currencyName}</span>
+              </div>
+            </div>
+          </div>
+          {/* Images */}
+          <div
+            className='flex space-x-5 overflow-x-auto overflow-y-hidden scrollbar-bg-white'
+            style={{ height: 'calc(100% + 6px)' }}
+          >
```
[the remark this answers is not in the document]
Comment (dbizhev) [reply src/features/members/MemberStrip.tsx:47]: @ArthurVogelsang I don't think it is possible to use tailwind css. If you don't have any idea, I'll mark it as resolved.

File: src/features/members/MemberCard.tsx [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+  followers: number | string
+  role?: string
+  isVerified: boolean
+}
+
+const MemberCard: React.FC<TMemberCard> = props => {
+  return (
+    <div className='px-5 py-7 bg-background-main rounded-2xl overflow-visible h-142px'>
+      <div className='flex space-x-4 relative'>
+        <img src={props.avatar} className='w-88px h-88px' />
```
[the remark this answers is not in the document]
Comment (dbizhev) [reply src/features/members/MemberCard.tsx]: @Dinisueldanov fixed

**Review by Dinisueldanov — APPROVED** [review]

**Review by ngvtuan — COMMENTED** [review]
ui issues 2

ui issues 3
`1` Can we add `unfollow` case ?

**Comment by dbizhev** [comment]
> ui issues 2
> 
> ui issues 3
> `1` Can we add `unfollow` case ?

@ngvtuan pushed the fixes.

File: src/common/components/ScrollBar/ScrollBar.tsx [hunk 10]
```diff
@@ -0,0 +1,48 @@
+import React, { ReactNode } from 'react'
+import SimpleBar from 'simplebar-react'
+import cn from 'classnames'
+
+// eslint-disable-next-line
+import 'simplebar/src/simplebar.css' //for some reason eslint was not accepting this
```
Comment (ngvtuan) [inline src/common/components/ScrollBar/ScrollBar.tsx:6]: I think you should use `import simplebar from 'simplebar/src/simplebar.css'` 

Please take a look at this sample https://github.com/pastelnetwork/pastel-electron-wallet/blob/master/src/features/receive/Receive.tsx (Line 12)

**Review by ngvtuan — COMMENTED** [review]
ui issues 1
It still has some minor issues. Could you please take a look and fix them?

**Comment by dbizhev** [comment]
> ui issues 1
> It still has some minor issues. Could you please take a look and fix them?

@ngvtuan There is no right padding in Figma design. If some content page has only left padding without right padding, it mean something goes wrong from the design side, do you agree?
They don't have pixel perfect design for sure now. They confirmed the design is still very dirty version and they need a lot of important fixes with design.

image

But currently our members directory page has same left and right padding from the front-end side and I'm sure we are making things better.

image

File: src/common/components/ScrollBar/ScrollBar.tsx (hunk 10 above)
Comment (dbizhev) [reply src/common/components/ScrollBar/ScrollBar.tsx:6]: @ngvtuan can you check here? https://www.npmjs.com/package/simplebar-react
image

Comment (dbizhev) [reply src/common/components/ScrollBar/ScrollBar.tsx:6]: @ngvtuan can you check here? https://github.com/Grsmto/simplebar/blob/master/examples/react/src/App.js#L2-L4

Comment (ngvtuan) [reply src/common/components/ScrollBar/ScrollBar.tsx:6]: @dbizhev I think the problem here is you are using the wrong path of the CSS when you try to import it Please try `import '../simplebar/dist/simplebar.min.css' `

Comment (dbizhev) [reply src/common/components/ScrollBar/ScrollBar.tsx:6]: @ngvtuan it's getting the error
image

Comment (dbizhev) [reply src/common/components/ScrollBar/ScrollBar.tsx:6]: @ngvtuan have you tested it on your end?

Comment (ngvtuan) [reply src/common/components/ScrollBar/ScrollBar.tsx:6]: @dbizhev let me check for the path. I think we need to include the `node_module` folder in the path. But It has another way to add the CSS file from the node_modules please take a look at here:  https://github.com/pastelnetwork/pastel-electron-wallet/blob/master/src/index.css (Line 31)

Comment (dbizhev) [reply src/common/components/ScrollBar/ScrollBar.tsx:6]: @ngvtuan why does it need?

Comment (dbizhev) [reply src/common/components/ScrollBar/ScrollBar.tsx]: @ngvtuan https://github.com/pastelnetwork/pastel-electron-wallet/pull/114/commits/90ddedfd8f8c7539ae36fa3945be2397859888c9#diff-85da366246340e911d7a0eb9153e64137a1b31f0686c74d9aef9ae4f032cb09eR9

**Review by ngvtuan — APPROVED** [review]
ui issues 1

I still found some issues related to the UI but as we already discussed in the group and we don't need to check the pixel-perfect for now. So it is ok to merge.
`1` `3` `4` belongs to another component and we will fix those components instead of this page. I will re-check again after those component completed
`2` Recheck after the design is completely done

**Pull request merged**